src/main/java/io/xdb/core/worker/WorkerService.java

```diff
@@ -41,8 +45,31 @@ public AsyncStateExecuteResponse handleAsyncStateExecute(final AsyncStateExecute
             .decode(request.getStateInput(), state.getInputType());
 
         // TODO
-        final StateDecision stateDecision = state.execute(input);
+        final io.xdb.core.state.StateDecision stateDecision = state.execute(input);
 
-        return new AsyncStateExecuteResponse().stateDecision(stateDecision);
+        return new AsyncStateExecuteResponse().stateDecision(toServerModel(request.getProcessType(), stateDecision));
+    }
+
+    private StateDecision toServerModel(final String processType, final io.xdb.core.state.StateDecision stateDecision) {
```

Comment:
nit: toApiModel or toInternalModels. the api is between server and SDKs, technically doesn't have to be owned by server